funnel/views/proposal.py

```diff
@@ -294,19 +287,24 @@ def proposal_view(profile, space, proposal):
                 flash(_("No such comment"), 'error')
             return redirect(proposal.url_for(), code=303)
     links = [Markup(linkify(unicode(escape(l)))) for l in proposal.links.replace('\r\n', '\n').split('\n') if l]
+
+    transitionform = None
     if not proposal.state.DRAFT:
-        statusform = ProposalStatusForm(status=proposal.state.value)
-    else:
-        statusform = None
+        transitionform = ProposalTransitionForm()
+        # fill up the form choices with the given proposal's transitions
+        transitionform.populate_transitions(proposal)
+
+    # considering the user an admin of the proposal if they have either one of the roles
+    user_is_admin = proposal.is_admin(g.user) or proposal.is_reviewer(g.user)
 
     proposal_move_form = None
-    if 'move-proposal' in space.permissions(g.user, g.permissions):
+    if user_is_admin:
```

Comment:
This is not wise. Don't assume permissions are granted because of a blanket check.